File: backend/app/api/v1/endpoints/muscle_groups.py
"""
Muscle Group API endpoints
"""
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select

from app.core.database import get_db
from app.models import MuscleGroup
from app.schemas.muscle_group import MuscleGroupBase, MuscleGroupListResponse
from fastapi import APIRouter, Depends

router = APIRouter()


@router.get("/", response_model=MuscleGroupListResponse)
async def list_muscle_groups(
    db: AsyncSession = Depends(get_db),
):
    """List all muscle groups"""
    query = select(MuscleGroup).order_by(MuscleGroup.level, MuscleGroup.name)
    
    result = await db.execute(query)
    muscle_groups = result.scalars().all()
    
    return MuscleGroupListResponse(
        muscle_groups=[MuscleGroupBase.model_validate(mg) for mg in muscle_groups]
    )

File: web/src/services/muscleGroups.ts
/**
 * Muscle Group API service
 */
import { apiClient } from './api'
import type { MuscleGroup } from './api'

export type { MuscleGroup }

export const muscleGroupApi = {
  /**
   * List all muscle groups
   */
  list: async (): Promise<MuscleGroup[]> => {
    const response = await apiClient.get<{ muscle_groups: MuscleGroup[] }>('/muscle-groups/')
    return response.data.muscle_groups
  },

  /**
   * Get a single muscle group by ID
   */
  get: async (id: number): Promise<MuscleGroup> => {
    const response = await apiClient.get<MuscleGroup>(`/muscle-groups/${id}`)
    return response.data
  },
}
